src/applications/edu-benefits/0994/components/HighTechEmploymentTypeView.jsx

```diff
@@ -0,0 +1,151 @@
+import React from 'react';
+import _ from 'lodash';
+import { TargetCheckboxWidget } from './TargetCheckboxWidget';
+
+const highTechEmploymentTypeIds = {
+  computerProgramming: {
+    value: 'computerProgramming',
+    title: 'Computer programming',
+  },
+  dataProcessing: {
+    value: 'dataProcessing',
+    title: 'Data processing',
+  },
+  computerSoftware: {
+    value: 'computerSoftware',
+    title: 'Computer software',
+  },
+  informationSciences: {
+    value: 'informationSciences',
+    title: 'Information sciences',
+  },
+  mediaApplication: {
+    value: 'mediaApplication',
+    title: 'Media application',
+  },
+  noneApply: {
+    value: 'noneApply',
+    title: 'None of these',
+  },
+};
+
+const setHighTechEmploymentType = (formData, property, value) => {
+  _.set(formData, property, value);
+};
+
+const isNoneApply = id => id === highTechEmploymentTypeIds.noneApply.value;
+
+const handleNoneApply = (id, formData) => {
+  if (isNoneApply(id)) {
+    setHighTechEmploymentType(
+      formData,
+      highTechEmploymentTypeIds.computerProgramming.value,
+      false,
+    );
+    setHighTechEmploymentType(
+      formData,
+      highTechEmploymentTypeIds.dataProcessing.value,
+      false,
+    );
+    setHighTechEmploymentType(
+      formData,
+      highTechEmploymentTypeIds.computerSoftware.value,
+      false,
+    );
+    setHighTechEmploymentType(
+      formData,
+      highTechEmploymentTypeIds.informationSciences.value,
+      false,
+    );
+    setHighTechEmploymentType(
+      formData,
+      highTechEmploymentTypeIds.mediaApplication.value,
+      false,
+    );
+  } else {
+    setHighTechEmploymentType(
+      formData,
+      highTechEmploymentTypeIds.noneApply.value,
+      false,
+    );
+  }
+};
+
+class HighTechEmploymentTypeView extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      ...props.formData,
+    };
+
+    this.handleChange = this.handleChange.bind(this);
+  }
+
+  handleChange(e) {
+    const { id, checked } = e;
+    const formData = { ...this.props.formData };
```

Comment:
You could change this line and the next two to:

```js
let formData;
if (isNoneApply(id)) {
   formData = { [highTechEmploymentTypeIds.noneApply.value]: checked };
} else {
   formData =  {
      ...formData,
      [highTechEmploymentTypeIds.noneApply.value]: false,
      [id]: checked
   };
}
```

This would let you get rid of a couple helper functions above. Our forms typically treat `undefined` and `false` as equivalent for these checkbox fields.